example/spring-data-mongodb/src/main/kotlin/com/github/inflab/example/spring/data/mongodb/repository/SetRepository.kt

```diff
@@ -0,0 +1,153 @@
+
+    data class Animal(val id: Long, val dogs: Long, val cats: Long)
+
+    data class Fruit(@Id val id: String, val item: String, val type: String)
+
+    /**
+     * @see <a href="https://www.mongodb.com/docs/manual/reference/operator/aggregation/set/#using-two--set-stages">Using Two $set Stages</a>
+     */
+    fun twoSetStages(): AggregationResults<ScoreWithTotal> {
+        val aggregation = aggregation {
+            set {
+                // TODO: add $sum expression
+                "totalHomework" set Sum.sumOf("homework")
+                "totalQuiz" set Sum.sumOf("quiz")
+            }
+            set {
+                "totalScore" set {
+                    add { of("totalHomework") and "totalQuiz" and "extraCredit" }
+                }
+            }
+        }
+
+        return mongoTemplate.aggregate<ScoreWithTotal>(aggregation, SCORES)
+    }
+
+    /**
+     * @see <a href="https://www.mongodb.com/docs/manual/reference/operator/aggregation/set/#adding-fields-to-an-embedded-document">Adding Fields to an Embedded Document</a>
+     */
+    fun setToEmbeddedDocument(): AggregationResults<Vehicle> {
+        val aggregation = aggregation {
+            set {
+                Vehicle::specs..Specs::fuelType set "unleaded"
+            }
+        }
+
+        return mongoTemplate.aggregate<Vehicle>(aggregation, VEHICLES)
+    }
+
+    /**
+     * @see <a href="https://www.mongodb.com/docs/manual/reference/operator/aggregation/set/#overwriting-an-existing-field">Overwriting an existing field</a>
+     */
+    fun overwriteAnExistingFieldWithValue(): AggregationResults<Animal> {
+        val aggregation = aggregation {
+            set {
+                Animal::cats set 20
+            }
+        }
+
+        return mongoTemplate.aggregate<Animal>(aggregation, ANIMALS)
+    }
+
+    /**
+     * @see <a href="https://www.mongodb.com/docs/manual/reference/operator/aggregation/set/#overwriting-an-existing-field">Overwriting an existing field</a>
+     */
+    fun overwriteAnExistingFieldWithField(): AggregationResults<Fruit> {
+        val aggregation = aggregation {
+            set {
+                Fruit::id set Fruit::item
+                Fruit::item set "fruit"
+            }
+        }
+
+        return mongoTemplate.aggregate<Fruit>(aggregation, FRUITS)
+    }
+
+    /**
+     * @see <a href="https://www.mongodb.com/docs/manual/reference/operator/aggregation/set/#add-element-to-an-array">Add Element to an Array</a>
+     */
+    fun addElementToAnArray(): AggregationResults<Score> {
+        val aggregation = aggregation {
+            match(Criteria.where(Score::id.toDotPath()).isEqualTo(1))
+            set {
+                // TODO: add $concatArrays expression
+                Score::homework set Document("\$concatArrays", listOf("\$${Score::homework.toDotPath()}", listOf(7)))
+            }
+        }
+
+        return mongoTemplate.aggregate<Score>(aggregation, SCORES)
+    }
```

Comment:
In example, we have to add `7` at the end of "homework" array.
```
db.scores.aggregate([
   { $match: { _id: 1 } },
   { $set: { homework: { $concatArrays: [ "$homework", [ 7 ] ] } } }
])
```

But, ConcatArrays not supported this. We can only concat value at the beginning maybe..

https://github.com/spring-projects/spring-data-mongodb/blob/69285151a0a8fc30b90cda06505bef973ea7151b/spring-data-mongodb/src/test/java/org/springframework/data/mongodb/core/aggregation/ArrayOperatorsUnitTests.java#L77-L82

Reply:
how about using `literal` expression. `literals(listof(7))`
but I can not sure if it works

Reply:
👍🏻👍🏻👍🏻👍🏻👍🏻👍🏻👍🏻👍🏻👍🏻👍🏻👍🏻👍🏻

Reply:
Hmm, I think it is better to implement `$concatArrays` without using ArrayOperator from spring.
It seems fine to leave it as it for now :)